apps/foundation-demo/src/app/design/molecules/sidebar/sidebar-viewport/sidebar-viewport.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, Output, EventEmitter, Input, ChangeDetectionStrategy } from '@angular/core';
+
+import { DaffSidebarMode } from '../helper/sidebar-mode';
+import { daffSidebarAnimations } from './sidebar-viewport.animation';
+
+@Component({
+  selector: 'daff-sidebar-viewport',
+  templateUrl: './sidebar-viewport.component.html',
+  styleUrls: ['./sidebar-viewport.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    daffSidebarAnimations.transformSidebar,
+    daffSidebarAnimations.transformContent,
+    daffSidebarAnimations.fadeBackdrop
+  ]
+})
+export class DaffSidebarViewportComponent{
+
+
+  _animationState: string = "void";
+
+  /**
+   * Property for the "opened" state of the sidebar
+   */
+  @Input()
+  get opened(): boolean { return this._opened; }
+  set opened(value: boolean) { 
+    if(this._opened == value){return;}
+    this._opened = value;
+    this._openedChanged.emit(value);
+    this._animationState = getAnimationState(value);
+  }
+
+  /**
+   * Output event to monitor when opened changes
+   */
+  
+  private _openedChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  /**
+   * Internal tracking variable for the state of sidebar viewport.
+   */
+  private _opened: boolean = false;
+
+  /**
+   * Event fired when the backdrop is clicked
+   * This is often used to close the sidebar
+   */
+  @Output() onBackdropClicked: EventEmitter<void> = new EventEmitter<void>();
+
+  _onBackdropClicked() : void {
+    this.onBackdropClicked.emit();
+  }
+
+  hasBackdrop() : boolean {
+    return this._opened && ( this.mode == DaffSidebarMode.OVER || this.mode == DaffSidebarMode.PUSH);
+  }
+
+  /**
+   * Input state for whether or not the backdrop is 
+   * "visible" to the human eye
+   */
+  @Input() backdropIsVisible: boolean = true;
+
+  /**
+   * The mode to put the sidebar in
+   */
+  @Input() mode: DaffSidebarMode = DaffSidebarMode.SIDE;
+
+}
+
+
+export const getAnimationState = (open : boolean = false, enabled: boolean = true) => {
```

Comment:
I think there's something wrong with how animation works with the `DaffSidebarMode.SIDE.side` choice. There's a weird delay before the sidebar disappears, and I think it's because it's supposed to be animating over a period of time, but whatever animations are happening in the background aren't actually happening to the sidebar.

Reply:
@lderrickable So, my intent with side mode (and this may be different from material) -- is that in "side" mode a sidebar can't be opened or closed. It merely sits alongside its content. (this is partially what led me to think we could call the `daff-sidebar-viewport` the `daff-sidebar-window`.

Reply:
But, I changed this, because there was indeed a bug as in `side` mode the sidebar is entirely hidden when opened -- this is fixed by 941a310c0fc731e69a8c1c06a5f21f93e0d40007